pkg/cephrbd/ceph_rbd_commands.go

```diff
@@ -0,0 +1,151 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cephrbd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/kubectl-rook-ceph/pkg"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	//k8s "k8s.io/client-go/kubernetes"
+
+	"k8s.io/client-go/kubernetes/scheme"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	//"k8s.io/client-go/rest"
+	restclient "k8s.io/client-go/rest"
+
+	log "github.com/sirupsen/logrus"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+func fail(msg string) {
+	fmt.Printf(msg)
+	os.Exit(1)
+}
+
+var (
+	OperatorNamespace    string // operator namespae
+	CephClusterNamespace string // Cephcluster namespace
+)
+
+func RunCommandInOperatorPod(cmd *cobra.Command, args []string, operatorNamespace, clusterNamespace string) {
+	var flagName []string
+	var flagValue []string
+
+	cmd.Flags().Parse(args)
+	cmd.Flags().Visit(func(f *pflag.Flag) {
+		flagName = append(flagName, f.Name)
+		flagValue = append(flagValue, f.Value.String())
+	})
+
+	// 1. Create Kubernetes Client
+	kubeconfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		clientcmd.NewDefaultClientConfigLoadingRules(),
+		&clientcmd.ConfigOverrides{},
+	)
+	rest, err := kubeconfig.ClientConfig()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Get Kubernetes Client
+	_, client := pkg.GetKubeClient()
+
+	opts := metav1.ListOptions{LabelSelector: fmt.Sprintf("app=%s", "rook-ceph-operator")}
+	list, err := client.Pods(operatorNamespace).List(context.TODO(), opts)
+	if err != nil || len(list.Items) == 0 {
+		log.Error("failed to get pod to exec into container")
+		log.Fatal(err)
+	}
+
+	get, err := client.Pods(operatorNamespace).Get(context.TODO(), list.Items[0].Name, metav1.GetOptions{})
+	if err != nil || len(get.Status.ContainerStatuses) == 0 {
+		log.Error("failed to get container")
+		log.Fatal(err)
+	}
+
+	ExecCmd(cmd, rest, get.Name, get.Namespace, get.Status.ContainerStatuses[0].Name, args, flagName, flagValue)
+}
+
+// ExecCmd exec command on specific pod and wait the command's output.
+func ExecCmd(command *cobra.Command, restconfig *restclient.Config, podName, podNameSpace, containerName string, args, flagName, flagValue []string) {
+	// Create a Kubernetes core/v1 client.
+	coreclient, err := corev1client.NewForConfig(restconfig)
+	if err != nil {
+		log.Fatal(err)
+	}
+	cmd := []string{
+		command.Use,
+	}
+	cmd = append(cmd, args...)
+
+	if len(flagName) != 0 {
+		for i := range flagName {
+			cmd = append(cmd, fmt.Sprint(flagName[i]), flagValue[i])
+		}
+	}
+	if cmd[0] == "ceph" {
+		cmd = append(cmd, "--connect-timeout=10", "--conf=/var/lib/rook/rook-ceph/rook-ceph.config")
+	} else {
+		cmd = append(cmd, "--conf=/var/lib/rook/rook-ceph/rook-ceph.config")
```

Comment:
Instead of duplicating this `--conf` option, can we factor out the `--connect-timeout` to a different condition?

Reply:
done